Monika After Story/game/script-farewells.rpy

```diff
@@ -13,101 +13,118 @@
 default persistent.mas_late_farewell = False
 
 init -1 python in mas_farewells:
-
-    # custom farewell functions
-    def selectFarewell():
+    import datetime
+    import store
+
+    def _filterFarewell(
+            ev,
+            curr_pri,
+            aff,
+            check_time,
+        ):
         """
-        Selects a farewell to be used. This evaluates rules and stuff
-        appropriately.
+        Filters a farewell for the given type, among other things.
+
+        IN:
+            ev - ev to filter
+            curr_pri - current loweset priority to compare to
+            aff - affection to use in aff_range comparisons
+            check_time - datetime to check against timed rules
 
         RETURNS:
-            a single farewell (as an Event) that we want to use
+            True if this ev passes the filter, False otherwise
         """
+        # NOTE: new rules:
+        #   eval in this order:
+        #   1. unlocked
+        #   2. not pooled
+        #   3. aff_range unlocked
+        #   4. conditional
+        #   5. priority (lower or same is True)
+        #   6. all rules
+        #       NOTE: this is never cleared. Please limit use of this
```

Comment:
this note applies to `conditional`. Additionally, since `conditional` is supposed to be a less likely to be used property, it should be last in the chain.